--- a/org.eclipse.core.resources/.classpath
+++ b/org.eclipse.core.resources/.classpath
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="src" path="src_ant"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/J2SE-1.4"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
 	<classpathentry kind="output" path="bin"/>
--- a/org.eclipse.core.resources/.project
+++ b/org.eclipse.core.resources/.project
@@ -3,6 +3,14 @@
 	<name>org.eclipse.core.resources</name>
 	<comment></comment>
 	<projects>
+	    <project>org.eclipse.core.jobs</project>
+	    <project>org.eclipse.osgi</project>
+	    <project>org.eclipse.equinox.registry</project>
+	    <project>org.eclipse.equinox.preferences</project>
+	    <project>org.eclipse.core.runtime</project>
+	    <project>org.eclipse.core.expressions</project>
+	    <project>org.eclipse.core.contenttype</project>
+	    <project>org.eclipse.core.runtime.compatibility.auth</project>
 	</projects>
 	<buildSpec>
 		<buildCommand>
--- a/org.eclipse.core.runtime/.project
+++ b/org.eclipse.core.runtime/.project
@@ -3,6 +3,12 @@
 	<name>org.eclipse.core.runtime</name>
 	<comment></comment>
 	<projects>
+    	<!--<project>org.eclipse.core.resources</project>-->
+    	<project>org.eclipse.equinox.app</project>
+    	<project>org.eclipse.core.runtime.compatibility.auth</project>
+    	<project>org.eclipse.osgi</project>
+    	<project>org.eclipse.core.contenttype</project>
+    	<project>org.eclipse.core.filesystem</project>
 	</projects>
 	<buildSpec>
 		<buildCommand>
--- a/org.eclipse.equinox.app/.project
+++ b/org.eclipse.equinox.app/.project
@@ -3,6 +3,7 @@
 	<name>org.eclipse.equinox.app</name>
 	<comment></comment>
 	<projects>
+	    <project>org.eclipse.osgi.services</project>
 	</projects>
 	<buildSpec>
 		<buildCommand>
--- a/org.eclipse.equinox.common/.project
+++ b/org.eclipse.equinox.common/.project
@@ -3,6 +3,7 @@
 	<name>org.eclipse.equinox.common</name>
 	<comment></comment>
 	<projects>
+	    <project>org.eclipse.osgi</project>
 	</projects>
 	<buildSpec>
 		<buildCommand>
--- a/org.eclipse.equinox.preferences/.project
+++ b/org.eclipse.equinox.preferences/.project
@@ -3,6 +3,7 @@
 	<name>org.eclipse.equinox.preferences</name>
 	<comment></comment>
 	<projects>
+	    <project>org.eclipse.equinox.common</project>
 	</projects>
 	<buildSpec>
 		<buildCommand>
--- a/org.eclipse.jdt.core/.project
+++ b/org.eclipse.jdt.core/.project
@@ -2,6 +2,11 @@
 <projectDescription>
 	<name>org.eclipse.jdt.core</name>
 	<comment></comment>
+	<projects>
+	    <project>org.eclipse.text</project>
+	    <project>org.eclipse.core.resources</project>
+	    <project>org.eclipse.core.runtime</project>
+	</projects>
 	<buildSpec>
 		<buildCommand>
 			<name>org.eclipse.jdt.core.javabuilder</name>
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/Buffer.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/Buffer.java
@@ -17,10 +17,7 @@ import java.util.ArrayList;
 
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IResource;
-import org.eclipse.core.runtime.CoreException;
-import org.eclipse.core.runtime.IProgressMonitor;
-import org.eclipse.core.runtime.ISafeRunnable;
-import org.eclipse.core.runtime.SafeRunner;
+import org.eclipse.core.runtime.*;
 import org.eclipse.core.runtime.content.IContentDescription;
 import org.eclipse.jdt.core.*;
 import org.eclipse.jdt.internal.core.util.Util;
@@ -257,7 +254,7 @@ protected void notifyChanged(final BufferChangedEvent event) {
 	if (listeners != null) {
 		for (int i = 0, size = listeners.size(); i < size; ++i) {
 			final IBufferChangedListener listener = (IBufferChangedListener) listeners.get(i);
-			SafeRunner.run(new ISafeRunnable() {
+			Platform.run(new ISafeRunnable() {
 				public void handleException(Throwable exception) {
 					Util.log(exception, "Exception occurred in listener of buffer change notification"); //$NON-NLS-1$
 				}
--- a/org.eclipse.text/.project
+++ b/org.eclipse.text/.project
@@ -3,6 +3,8 @@
 	<name>org.eclipse.text</name>
 	<comment></comment>
 	<projects>
+	    <project>org.eclipse.equinox.common</project>
+	    <project>org.eclipse.core.commands</project>
 	</projects>
 	<buildSpec>
 		<buildCommand>
--- a/org.eclipse.text/src/org/eclipse/jface/text/templates/GlobalTemplateVariables.java
+++ b/org.eclipse.text/src/org/eclipse/jface/text/templates/GlobalTemplateVariables.java
@@ -11,9 +11,6 @@
  *******************************************************************************/
 package org.eclipse.jface.text.templates;
 
-import com.ibm.icu.text.DateFormat;
-import com.ibm.icu.util.Calendar;
-
 /**
  * Global variables which are available in any context.
  * <p>
@@ -114,7 +111,7 @@ public class GlobalTemplateVariables {
 			super("date", TextTemplateMessages.getString("GlobalVariables.variable.description.date")); //$NON-NLS-1$ //$NON-NLS-2$
 		}
 		protected String resolve(TemplateContext context) {
-			return DateFormat.getDateInstance().format(new java.util.Date());
+			throw new RuntimeException("Cannot resolve DATE (stubbed out)!");
 		}
 	}
 
@@ -129,7 +126,7 @@ public class GlobalTemplateVariables {
 			super("year", TextTemplateMessages.getString("GlobalVariables.variable.description.year")); //$NON-NLS-1$ //$NON-NLS-2$
 		}
 		protected String resolve(TemplateContext context) {
-			return Integer.toString(Calendar.getInstance().get(Calendar.YEAR));
+			throw new RuntimeException("Cannot resolve YEAR (stubbed out)!");
 		}
 	}
 
@@ -148,7 +145,7 @@ public class GlobalTemplateVariables {
 		 * {@inheritDoc}
 		 */
 		protected String resolve(TemplateContext context) {
-			return DateFormat.getTimeInstance().format(new java.util.Date());
+			throw new RuntimeException("Cannot resolve TIME (stubbed out)!");
 		}
 	}
 
--- a/org.eclipse.text/src/org/eclipse/jface/text/templates/TextTemplateMessages.java
+++ b/org.eclipse.text/src/org/eclipse/jface/text/templates/TextTemplateMessages.java
@@ -10,7 +10,6 @@
  *******************************************************************************/
 package org.eclipse.jface.text.templates;
 
-import com.ibm.icu.text.MessageFormat;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
 
@@ -34,11 +33,11 @@ class TextTemplateMessages {
 	}
 
 	public static String getFormattedString(String key, Object arg) {
-		return MessageFormat.format(getString(key), new Object[] { arg });
+		throw new RuntimeException("Cannot format string (stubbed out)!");
 	}
 
 
 	public static String getFormattedString(String key, Object[] args) {
-		return MessageFormat.format(getString(key), args);
+		throw new RuntimeException("Cannot format string (stubbed out)!");
 	}
 }
--- a/org.eclipse.text/src/org/eclipse/text/edits/TextEditMessages.java
+++ b/org.eclipse.text/src/org/eclipse/text/edits/TextEditMessages.java
@@ -10,7 +10,6 @@
  *******************************************************************************/
 package org.eclipse.text.edits;
 
-import com.ibm.icu.text.MessageFormat;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
 
@@ -36,6 +35,6 @@ class TextEditMessages {
 	}
 
 	public static String getFormattedString(String key, Object[] args) {
-		return MessageFormat.format(getString(key), args);
+		throw new RuntimeException("cannot get formatted string (STUBBED OUT)");
 	}
 }
--- a/org.eclipse.text/src/org/eclipse/text/undo/UndoMessages.java
+++ b/org.eclipse.text/src/org/eclipse/text/undo/UndoMessages.java
@@ -10,7 +10,6 @@
  *******************************************************************************/
 package org.eclipse.text.undo;
 
-import com.ibm.icu.text.MessageFormat;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
 
@@ -41,6 +40,6 @@ final class UndoMessages {
 	}
 
 	public static String getFormattedString(String key, Object[] args) {
-		return MessageFormat.format(getString(key), args);
+		throw new RuntimeException("stubbed out string format");
 	}
 }
