--- a/native/jni/native-lib/cpio.c
+++ b/native/jni/native-lib/cpio.c
@@ -482,6 +482,8 @@ int cpio_checkType (const char *filename, jint *entryType)
 {
   struct stat statbuf;
 
+  *entryType = CPFILE_UNKNOWN;
+
   if (stat(filename, &statbuf) < 0)
     return errno;
 
--- a/native/jni/native-lib/cpio.h
+++ b/native/jni/native-lib/cpio.h
@@ -71,6 +71,7 @@ JNIEXPORT int cpio_closeOnExec(int fd);
 
 #define CPFILE_FILE 0
 #define CPFILE_DIRECTORY 1
+#define CPFILE_UNKNOWN 2
 
 JNIEXPORT int cpio_setFileReadonly (const char *filename);
 JNIEXPORT int cpio_chmod (const char *filename, int permissions);
