--- a/configure
+++ b/configure
@@ -24470,7 +24470,7 @@ fi
 test "x$JAVA" = x && as_fn_error $? "no acceptable Java virtual machine found in \$PATH" "$LINENO" 5
 
 
-ECJ_OPTS="-warn:-deprecation,serial,unusedImport"
+ECJ_OPTS="-warn:-deprecation,unusedImport"
 JAVAC_OPTS="-Xlint:unchecked,cast,divzero,empty,finally,overrides"
 GCJ_OPTS="-g"
 if test "x$JAVAPREFIX" = x; then
--- a/m4/ac_prog_javac.m4
+++ b/m4/ac_prog_javac.m4
@@ -36,7 +36,7 @@ dnl Modified to remove jikes by Andrew John Hughes on 2008-02-11
 
 AC_DEFUN_ONCE([AC_PROG_JAVAC],[
 AC_REQUIRE([AC_EXEEXT])dnl
-ECJ_OPTS="-warn:-deprecation,serial,unusedImport"
+ECJ_OPTS="-warn:-deprecation,unusedImport"
 JAVAC_OPTS="-Xlint:unchecked,cast,divzero,empty,finally,overrides"
 GCJ_OPTS="-g"
 if test "x$JAVAPREFIX" = x; then
